src/Stripe.net/Infrastructure/Middleware/RequestStringBuilder.cs

```diff
@@ -1,61 +1,416 @@
+        }
+
+        /// <summary>
+        /// Creates the HTTP query string for a given options class.
+        /// </summary>
+        /// <paramref name="requestString">The string to which the query string will be appended.</param>
+        /// <param name="options">The options class for which to create the query string.</param>
+        public static void CreateQuery(ref string requestString, INestedOptions options)
+        {
+            List<Parameter> flatParams = FlattenParams(options);
+
+            foreach (Parameter param in flatParams)
+            {
+                RequestStringBuilder.ApplyParameterToRequestString(ref requestString, param.Key, param.Value);
+            }
+        }
+
+        /// <summary>
+        /// Returns a list of parameters for a given options class. If the class contains
+        /// containers (e.g. dictionaries, lists, arrays or other options classes), the parameters
+        /// will be computed recursively and a flat list will be returned.
+        /// </summary>
+        /// <param name="options">The options class for which to create the list of parameters.</param>
+        /// <returns>The list of parameters</returns>
+        private static List<Parameter> FlattenParams(INestedOptions options)
+        {
+            return FlattenParamsOptions(options, null);
+        }
+
+        /// <summary>
+        /// Returns a list of parameters for a given options class. If a key prefix is provided, the
+        /// keys for the new parameters will be nested under the key prefix. E.g. if the key prefix
+        /// `foo` is passed and the options class contains a parameter `bar`, then a parameter
+        /// with key `foo[bar]` will be returned.
+        /// </summary>
+        /// <param name="options">The options class for which to create the list of parameters.</param>
+        /// <param name="keyPrefix">The key under which new keys should be nested, if any.</param>
+        /// <returns>The list of parameters</returns>
+        private static List<Parameter> FlattenParamsOptions(INestedOptions options, string keyPrefix)
+        {
+            List<Parameter> flatParams = new List<Parameter>();
+            if (options == null)
+            {
+                return flatParams;
+            }
+
+            foreach (var property in options.GetType().GetRuntimeProperties())
+            {
+                var value = property.GetValue(options, null);
+                if (value == null)
+                {
+                    continue;
+                }
+
+                var attribute = property.GetCustomAttribute<JsonPropertyAttribute>();
+                if (attribute == null)
+                {
+                    continue;
+                }
+
+                string key = attribute.PropertyName;
+                string newPrefix = NewPrefix(key, keyPrefix);
+
+                flatParams.AddRange(FlattenParamsValue(value, newPrefix));
```

Comment:
I don't think this is necessary. We compute a list of parameters, not a dictionary, so there's no possibility of overwriting. If a key is present more than once in the list, it will be encoded and sent in the request, and the API / stripe-mock would complain.